### PR TITLE
r/s3_bucket: read-only `server_side_encryption_configuration` argument

### PR DESCRIPTION
**Pull request by anGie44**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates #4418
Relates #20433

Output from acceptance testing:

```
--- PASS: TestAccS3Bucket_Basic_acceleration (171.29s)
--- PASS: TestAccS3Bucket_Basic_basic (97.47s)
--- PASS: TestAccS3Bucket_Basic_emptyString (97.74s)
--- PASS: TestAccS3Bucket_Basic_forceDestroy (85.81s)
--- PASS: TestAccS3Bucket_Basic_forceDestroyWithEmptyPrefixes (80.52s)
--- PASS: TestAccS3Bucket_Basic_forceDestroyWithObjectLockEnabled (76.97s)
--- PASS: TestAccS3Bucket_Basic_generatedName (96.27s)
--- PASS: TestAccS3Bucket_Basic_namePrefix (94.31s)
--- PASS: TestAccS3Bucket_Basic_requestPayer (165.41s)
--- PASS: TestAccS3Bucket_Basic_shouldFailNotFound (72.06s)
--- PASS: TestAccS3Bucket_Manage_MfaDeleteDisabled (96.84s)
--- PASS: TestAccS3Bucket_Manage_lifecycleBasic (213.92s)
--- PASS: TestAccS3Bucket_Manage_lifecycleExpireMarkerOnly (176.00s)
--- PASS: TestAccS3Bucket_Manage_lifecycleRuleAbortIncompleteMultipartUploadDaysNoExpiration (103.06s)
--- PASS: TestAccS3Bucket_Manage_lifecycleRuleExpirationEmptyBlock (90.03s)
--- PASS: TestAccS3Bucket_Manage_objectLock (120.46s)
--- PASS: TestAccS3Bucket_Manage_versioning (174.97s)
--- PASS: TestAccS3Bucket_Manage_versioningAndMfaDeleteDisabled (98.13s)
--- PASS: TestAccS3Bucket_Manage_versioningDisabled (100.32s)
--- PASS: TestAccS3Bucket_Replication_RTC_valid (211.19s)
--- PASS: TestAccS3Bucket_Replication_basic (269.60s)
--- PASS: TestAccS3Bucket_Replication_expectVersioningValidationError (76.84s)
--- PASS: TestAccS3Bucket_Replication_multipleDestinationsEmptyFilter (161.17s)
--- PASS: TestAccS3Bucket_Replication_multipleDestinationsNonEmptyFilter (163.99s)
--- PASS: TestAccS3Bucket_Replication_ruleDestinationAccessControlTranslation (202.43s)
--- PASS: TestAccS3Bucket_Replication_ruleDestinationAddAccessControlTranslation (197.38s)
--- PASS: TestAccS3Bucket_Replication_schemaV2 (240.61s)
--- PASS: TestAccS3Bucket_Replication_schemaV2SameRegion (103.78s)
--- PASS: TestAccS3Bucket_Replication_twoDestination (162.30s)
--- PASS: TestAccS3Bucket_Replication_withoutPrefix (125.83s)
--- PASS: TestAccS3Bucket_Replication_withoutStorageClass (156.04s)
--- PASS: TestAccS3Bucket_Security_aclToGrant (157.96s)
--- PASS: TestAccS3Bucket_Security_corsDelete (87.52s)
--- PASS: TestAccS3Bucket_Security_corsEmptyOrigin (101.25s)
--- PASS: TestAccS3Bucket_Security_corsUpdate (179.05s)
--- PASS: TestAccS3Bucket_Security_grantToACL (152.71s)
--- PASS: TestAccS3Bucket_Security_logging (104.98s)
--- PASS: TestAccS3Bucket_Security_policy (243.01s)
--- PASS: TestAccS3Bucket_Security_updateACL (165.88s)
--- PASS: TestAccS3Bucket_Security_updateGrant (241.55s)
--- PASS: TestAccS3Bucket_Tags_basic (98.37s)
--- PASS: TestAccS3Bucket_Tags_ignoreTags (151.04s)
--- PASS: TestAccS3Bucket_Tags_withNoSystemTags (310.78s)
--- PASS: TestAccS3Bucket_Tags_withSystemTags (357.39s)
--- PASS: TestAccS3Bucket_Web_redirect (241.15s)
--- PASS: TestAccS3Bucket_Web_routingRules (168.63s)
--- PASS: TestAccS3Bucket_Web_simple (241.34s)
```

### CODE REVIEW

**Comment by anGie44** [comment]
Preceded by https://github.com/hashicorp/terraform-provider-aws/pull/22609

File: internal/service/s3/bucket_test.go [hunk 1]
```diff
@@ -791,133 +791,6 @@ func TestAccS3Bucket_Web_routingRules(t *testing.T) {
 	})
 }
 
-func TestAccS3Bucket_Security_enableDefaultEncryptionWhenTypical(t *testing.T) {
```
Comment (anGie44) [inline internal/service/s3/bucket_test.go:794]: Tests ported over to #22609

File: .changelog/22605.txt [hunk 2]
```diff
@@ -0,0 +1,3 @@
+```release-note:note
```
Comment (anGie44) [inline .changelog/22605.txt:1]:
```suggestion
```release-note:breaking-change
```

**Comment by ewbankkit** [comment]
If I attempt a `terraform plan` on an S3 bucket created in a previous version of the provider I get:

```console
aws_s3_bucket.test: Refreshing state... [id=ewbankkit-test-default-s3-bucket-encryption-001]
╷
│ Error: error setting server_side_encryption_configuration: Invalid address to set: []string{"server_side_encryption_configuration", "0", "rules"}
│ 
│   with aws_s3_bucket.test,
│   on main.tf line 3, in resource "aws_s3_bucket" "test":
│    3: resource "aws_s3_bucket" "test" {
```
In `aws_s3_bucket` the nested attribute is `rule`, not `rules`.

File: internal/service/s3/bucket.go [hunk 3]
```diff
@@ -2371,24 +2298,41 @@ func resourceBucketLifecycleUpdate(conn *s3.S3, d *schema.ResourceData) error {
...
-			r["apply_server_side_encryption_by_default"] = []map[string]interface{}{d}
-			r["bucket_key_enabled"] = aws.BoolValue(v.BucketKeyEnabled)
-			rules = append(rules, r)
+func flattenServerSideEncryptionConfiguration(c *s3.ServerSideEncryptionConfiguration) []interface{} {
+	if c == nil {
+		return []interface{}{}
+	}
+
+	m := map[string]interface{}{
+		"rules": flattenServerSideEncryptionConfigurationRules(c.Rules),
```
Comment (anGie44) [inline internal/service/s3/bucket.go:2307]: source of error :/ 
```suggestion
		"rule": flattenServerSideEncryptionConfigurationRules(c.Rules),
```

File: website/docs/r/s3_bucket.html.markdown [hunk 4]
```diff
@@ -305,25 +305,8 @@ resource "aws_s3_bucket" "source" {
...
-    rule {
-      apply_server_side_encryption_by_default {
-        kms_master_key_id = aws_kms_key.mykey.arn
-        sse_algorithm     = "aws:kms"
-      }
-    }
-  }
-}
-```
+The `server_side_encryption_configuration` argument is read-only as of version 4.0.
```
Comment (anGie44) [inline website/docs/r/s3_bucket.html.markdown:308]:
```suggestion
The `server_side_encryption_configuration` argument is read-only as of version 4.0 of the Terraform AWS Provider.
```

File: internal/service/s3/bucket.go [hunk 5]
```diff
@@ -1291,16 +1287,14 @@ func resourceBucketRead(d *schema.ResourceData, meta interface{}) error {
...
-	serverSideEncryptionConfiguration := make([]map[string]interface{}, 0)
-	if encryption, ok := encryptionResponse.(*s3.GetBucketEncryptionOutput); ok && encryption.ServerSideEncryptionConfiguration != nil {
-		serverSideEncryptionConfiguration = flattenServerSideEncryptionConfiguration(encryption.ServerSideEncryptionConfiguration)
-	}
-	if err := d.Set("server_side_encryption_configuration", serverSideEncryptionConfiguration); err != nil {
-		return fmt.Errorf("error setting server_side_encryption_configuration: %s", err)
+	if encryption, ok := encryptionResponse.(*s3.GetBucketEncryptionOutput); ok {
+		if err := d.Set("server_side_encryption_configuration", flattenServerSideEncryptionConfiguration(encryption.ServerSideEncryptionConfiguration)); err != nil {
+			return fmt.Errorf("error setting server_side_encryption_configuration: %w", err)
+		}
```
Comment (ewbankkit) [inline internal/service/s3/bucket.go:1297]: I think you'll need `d.Set("server_side_encryption_configuration", nil)` in an `else` branch here as previous logic was setting this attribute even if `encryptionResponse.(*s3.GetBucketEncryptionOutput)` was `nil`.

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc TESTARGS='-run=TestAccS3Bucket_' PKG=s3 ACCTEST_PARALLELISM=5
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/s3/... -v -count 1 -parallel 5  -run=TestAccS3Bucket_ -timeout 180m
=== RUN   TestAccS3Bucket_Basic_basic
=== PAUSE TestAccS3Bucket_Basic_basic
=== RUN   TestAccS3Bucket_Basic_emptyString
=== PAUSE TestAccS3Bucket_Basic_emptyString
=== RUN   TestAccS3Bucket_Tags_withNoSystemTags
=== PAUSE TestAccS3Bucket_Tags_withNoSystemTags
=== RUN   TestAccS3Bucket_Tags_withSystemTags
=== PAUSE TestAccS3Bucket_Tags_withSystemTags
=== RUN   TestAccS3Bucket_Tags_ignoreTags
=== PAUSE TestAccS3Bucket_Tags_ignoreTags
=== RUN   TestAccS3Bucket_Tags_basic
=== PAUSE TestAccS3Bucket_Tags_basic
=== RUN   TestAccS3Bucket_Basic_namePrefix
=== PAUSE TestAccS3Bucket_Basic_namePrefix
=== RUN   TestAccS3Bucket_Basic_generatedName
=== PAUSE TestAccS3Bucket_Basic_generatedName
=== RUN   TestAccS3Bucket_Basic_acceleration
=== PAUSE TestAccS3Bucket_Basic_acceleration
=== RUN   TestAccS3Bucket_Basic_requestPayer
=== PAUSE TestAccS3Bucket_Basic_requestPayer
=== RUN   TestAccS3Bucket_Security_policy
=== PAUSE TestAccS3Bucket_Security_policy
=== RUN   TestAccS3Bucket_Security_updateACL
=== PAUSE TestAccS3Bucket_Security_updateACL
=== RUN   TestAccS3Bucket_Security_updateGrant
=== PAUSE TestAccS3Bucket_Security_updateGrant
=== RUN   TestAccS3Bucket_Security_aclToGrant
=== PAUSE TestAccS3Bucket_Security_aclToGrant
=== RUN   TestAccS3Bucket_Security_grantToACL
=== PAUSE TestAccS3Bucket_Security_grantToACL
=== RUN   TestAccS3Bucket_Web_simple
=== PAUSE TestAccS3Bucket_Web_simple
=== RUN   TestAccS3Bucket_Web_redirect
=== PAUSE TestAccS3Bucket_Web_redirect
=== RUN   TestAccS3Bucket_Web_routingRules
=== PAUSE TestAccS3Bucket_Web_routingRules
=== RUN   TestAccS3Bucket_Basic_shouldFailNotFound
=== PAUSE TestAccS3Bucket_Basic_shouldFailNotFound
=== RUN   TestAccS3Bucket_Manage_versioning
=== PAUSE TestAccS3Bucket_Manage_versioning
=== RUN   TestAccS3Bucket_Manage_versioningDisabled
=== PAUSE TestAccS3Bucket_Manage_versioningDisabled
=== RUN   TestAccS3Bucket_Manage_MfaDeleteDisabled
=== PAUSE TestAccS3Bucket_Manage_MfaDeleteDisabled
=== RUN   TestAccS3Bucket_Manage_versioningAndMfaDeleteDisabled
=== PAUSE TestAccS3Bucket_Manage_versioningAndMfaDeleteDisabled
=== RUN   TestAccS3Bucket_Security_corsUpdate
=== PAUSE TestAccS3Bucket_Security_corsUpdate
=== RUN   TestAccS3Bucket_Security_corsDelete
=== PAUSE TestAccS3Bucket_Security_corsDelete
=== RUN   TestAccS3Bucket_Security_corsEmptyOrigin
=== PAUSE TestAccS3Bucket_Security_corsEmptyOrigin
=== RUN   TestAccS3Bucket_Security_logging
=== PAUSE TestAccS3Bucket_Security_logging
=== RUN   TestAccS3Bucket_Manage_lifecycleBasic
=== PAUSE TestAccS3Bucket_Manage_lifecycleBasic
=== RUN   TestAccS3Bucket_Manage_lifecycleExpireMarkerOnly
=== PAUSE TestAccS3Bucket_Manage_lifecycleExpireMarkerOnly
=== RUN   TestAccS3Bucket_Manage_lifecycleRuleExpirationEmptyBlock
=== PAUSE TestAccS3Bucket_Manage_lifecycleRuleExpirationEmptyBlock
=== RUN   TestAccS3Bucket_Manage_lifecycleRuleAbortIncompleteMultipartUploadDaysNoExpiration
=== PAUSE TestAccS3Bucket_Manage_lifecycleRuleAbortIncompleteMultipartUploadDaysNoExpiration
=== RUN   TestAccS3Bucket_Replication_basic
=== PAUSE TestAccS3Bucket_Replication_basic
=== RUN   TestAccS3Bucket_Replication_multipleDestinationsEmptyFilter
=== PAUSE TestAccS3Bucket_Replication_multipleDestinationsEmptyFilter
=== RUN   TestAccS3Bucket_Replication_multipleDestinationsNonEmptyFilter
=== PAUSE TestAccS3Bucket_Replication_multipleDestinationsNonEmptyFilter
=== RUN   TestAccS3Bucket_Replication_twoDestination
=== PAUSE TestAccS3Bucket_Replication_twoDestination
=== RUN   TestAccS3Bucket_Replication_ruleDestinationAccessControlTranslation
=== PAUSE TestAccS3Bucket_Replication_ruleDestinationAccessControlTranslation
=== RUN   TestAccS3Bucket_Replication_ruleDestinationAddAccessControlTranslation
=== PAUSE TestAccS3Bucket_Replication_ruleDestinationAddAccessControlTranslation
=== RUN   TestAccS3Bucket_Replication_withoutStorageClass
=== PAUSE TestAccS3Bucket_Replication_withoutStorageClass
=== RUN   TestAccS3Bucket_Replication_expectVersioningValidationError
=== PAUSE TestAccS3Bucket_Replication_expectVersioningValidationError
=== RUN   TestAccS3Bucket_Replication_withoutPrefix
=== PAUSE TestAccS3Bucket_Replication_withoutPrefix
=== RUN   TestAccS3Bucket_Replication_schemaV2
=== PAUSE TestAccS3Bucket_Replication_schemaV2
=== RUN   TestAccS3Bucket_Replication_schemaV2SameRegion
=== PAUSE TestAccS3Bucket_Replication_schemaV2SameRegion
=== RUN   TestAccS3Bucket_Replication_RTC_valid
=== PAUSE TestAccS3Bucket_Replication_RTC_valid
=== RUN   TestAccS3Bucket_Manage_objectLock
=== PAUSE TestAccS3Bucket_Manage_objectLock
=== RUN   TestAccS3Bucket_Basic_forceDestroy
=== PAUSE TestAccS3Bucket_Basic_forceDestroy
=== RUN   TestAccS3Bucket_Basic_forceDestroyWithEmptyPrefixes
=== PAUSE TestAccS3Bucket_Basic_forceDestroyWithEmptyPrefixes
=== RUN   TestAccS3Bucket_Basic_forceDestroyWithObjectLockEnabled
=== PAUSE TestAccS3Bucket_Basic_forceDestroyWithObjectLockEnabled
=== CONT  TestAccS3Bucket_Basic_basic
=== CONT  TestAccS3Bucket_Security_corsDelete
=== CONT  TestAccS3Bucket_Security_updateGrant
=== CONT  TestAccS3Bucket_Security_corsUpdate
=== CONT  TestAccS3Bucket_Manage_versioningAndMfaDeleteDisabled
--- PASS: TestAccS3Bucket_Security_corsDelete (42.35s)
=== CONT  TestAccS3Bucket_Manage_MfaDeleteDisabled
--- PASS: TestAccS3Bucket_Basic_basic (51.40s)
=== CONT  TestAccS3Bucket_Manage_versioningDisabled
--- PASS: TestAccS3Bucket_Manage_versioningAndMfaDeleteDisabled (51.41s)
=== CONT  TestAccS3Bucket_Manage_versioning
--- PASS: TestAccS3Bucket_Security_corsUpdate (90.38s)
=== CONT  TestAccS3Bucket_Basic_shouldFailNotFound
--- PASS: TestAccS3Bucket_Manage_MfaDeleteDisabled (48.06s)
=== CONT  TestAccS3Bucket_Web_routingRules
--- PASS: TestAccS3Bucket_Manage_versioningDisabled (41.85s)
=== CONT  TestAccS3Bucket_Web_redirect
--- PASS: TestAccS3Bucket_Security_updateGrant (131.35s)
=== CONT  TestAccS3Bucket_Web_simple
--- PASS: TestAccS3Bucket_Manage_versioning (88.63s)
=== CONT  TestAccS3Bucket_Security_grantToACL
--- PASS: TestAccS3Bucket_Web_routingRules (90.78s)
=== CONT  TestAccS3Bucket_Security_aclToGrant
--- PASS: TestAccS3Bucket_Web_redirect (123.38s)
=== CONT  TestAccS3Bucket_Replication_ruleDestinationAddAccessControlTranslation
--- PASS: TestAccS3Bucket_Security_grantToACL (76.59s)
=== CONT  TestAccS3Bucket_Basic_forceDestroyWithObjectLockEnabled
=== CONT  TestAccS3Bucket_Basic_shouldFailNotFound
    bucket_test.go:801: Step 1/1 error: Error running post-apply refresh: exit status 1
        
        Error: error getting S3 Bucket CORS configuration: NoSuchBucket: The specified bucket does not exist
                status code: 404, request id: H1E3V3P2J1BR70Y2, host id: 5nWlxU+rJbBQ3LojPRKKMPMUuMTnxLRVH+j/mAGnIv92/0dGaolHXhZL/jPkv40WmiBnR8jHw18=
        
          with aws_s3_bucket.bucket,
          on terraform_plugin_test.tf line 2, in resource "aws_s3_bucket" "bucket":
           2: resource "aws_s3_bucket" "bucket" {
        
--- FAIL: TestAccS3Bucket_Basic_shouldFailNotFound (145.96s)
=== CONT  TestAccS3Bucket_Basic_forceDestroyWithEmptyPrefixes
--- PASS: TestAccS3Bucket_Security_aclToGrant (72.27s)
=== CONT  TestAccS3Bucket_Basic_forceDestroy
--- PASS: TestAccS3Bucket_Web_simple (122.67s)
=== CONT  TestAccS3Bucket_Manage_objectLock
--- PASS: TestAccS3Bucket_Basic_forceDestroyWithObjectLockEnabled (41.99s)
=== CONT  TestAccS3Bucket_Replication_RTC_valid
--- PASS: TestAccS3Bucket_Basic_forceDestroyWithEmptyPrefixes (38.60s)
=== CONT  TestAccS3Bucket_Replication_schemaV2SameRegion
--- PASS: TestAccS3Bucket_Basic_forceDestroy (38.88s)
=== CONT  TestAccS3Bucket_Replication_schemaV2
--- PASS: TestAccS3Bucket_Replication_schemaV2SameRegion (61.98s)
=== CONT  TestAccS3Bucket_Replication_withoutPrefix
--- PASS: TestAccS3Bucket_Manage_objectLock (83.00s)
=== CONT  TestAccS3Bucket_Replication_expectVersioningValidationError
--- PASS: TestAccS3Bucket_Replication_expectVersioningValidationError (37.94s)
=== CONT  TestAccS3Bucket_Replication_withoutStorageClass
--- PASS: TestAccS3Bucket_Replication_ruleDestinationAddAccessControlTranslation (200.33s)
=== CONT  TestAccS3Bucket_Manage_lifecycleRuleAbortIncompleteMultipartUploadDaysNoExpiration
--- PASS: TestAccS3Bucket_Replication_withoutPrefix (89.14s)
=== CONT  TestAccS3Bucket_Replication_ruleDestinationAccessControlTranslation
--- PASS: TestAccS3Bucket_Replication_withoutStorageClass (54.10s)
=== CONT  TestAccS3Bucket_Replication_twoDestination
--- PASS: TestAccS3Bucket_Manage_lifecycleRuleAbortIncompleteMultipartUploadDaysNoExpiration (46.40s)
=== CONT  TestAccS3Bucket_Replication_multipleDestinationsNonEmptyFilter
--- PASS: TestAccS3Bucket_Replication_RTC_valid (207.12s)
=== CONT  TestAccS3Bucket_Replication_multipleDestinationsEmptyFilter
--- PASS: TestAccS3Bucket_Replication_twoDestination (70.12s)
=== CONT  TestAccS3Bucket_Replication_basic
--- PASS: TestAccS3Bucket_Replication_multipleDestinationsNonEmptyFilter (56.58s)
=== CONT  TestAccS3Bucket_Manage_lifecycleBasic
--- PASS: TestAccS3Bucket_Replication_schemaV2 (239.36s)
=== CONT  TestAccS3Bucket_Manage_lifecycleRuleExpirationEmptyBlock
--- PASS: TestAccS3Bucket_Replication_ruleDestinationAccessControlTranslation (110.31s)
=== CONT  TestAccS3Bucket_Manage_lifecycleExpireMarkerOnly
--- PASS: TestAccS3Bucket_Replication_multipleDestinationsEmptyFilter (91.59s)
=== CONT  TestAccS3Bucket_Security_logging
--- PASS: TestAccS3Bucket_Manage_lifecycleRuleExpirationEmptyBlock (33.98s)
=== CONT  TestAccS3Bucket_Security_corsEmptyOrigin
--- PASS: TestAccS3Bucket_Security_corsEmptyOrigin (40.56s)
=== CONT  TestAccS3Bucket_Basic_namePrefix
--- PASS: TestAccS3Bucket_Security_logging (53.68s)
=== CONT  TestAccS3Bucket_Security_updateACL
--- PASS: TestAccS3Bucket_Manage_lifecycleExpireMarkerOnly (77.45s)
=== CONT  TestAccS3Bucket_Security_policy
--- PASS: TestAccS3Bucket_Manage_lifecycleBasic (109.23s)
=== CONT  TestAccS3Bucket_Basic_requestPayer
--- PASS: TestAccS3Bucket_Basic_namePrefix (44.37s)
=== CONT  TestAccS3Bucket_Basic_acceleration
--- PASS: TestAccS3Bucket_Security_updateACL (77.22s)
=== CONT  TestAccS3Bucket_Basic_generatedName
--- PASS: TestAccS3Bucket_Replication_basic (211.37s)
=== CONT  TestAccS3Bucket_Tags_withSystemTags
--- PASS: TestAccS3Bucket_Basic_requestPayer (85.73s)
=== CONT  TestAccS3Bucket_Tags_basic
--- PASS: TestAccS3Bucket_Security_policy (118.55s)
=== CONT  TestAccS3Bucket_Tags_ignoreTags
--- PASS: TestAccS3Bucket_Basic_acceleration (81.83s)
=== CONT  TestAccS3Bucket_Tags_withNoSystemTags
--- PASS: TestAccS3Bucket_Basic_generatedName (49.45s)
=== CONT  TestAccS3Bucket_Basic_emptyString
--- PASS: TestAccS3Bucket_Tags_basic (49.95s)
--- PASS: TestAccS3Bucket_Basic_emptyString (45.50s)
--- PASS: TestAccS3Bucket_Tags_ignoreTags (81.57s)
--- PASS: TestAccS3Bucket_Tags_withNoSystemTags (163.84s)
--- PASS: TestAccS3Bucket_Tags_withSystemTags (207.32s)
FAIL
FAIL    github.com/hashicorp/terraform-provider-aws/internal/service/s3 925.095s
FAIL
make: *** [testacc] Error 1
```

**Comment by anGie44** [comment]
Hmm i haven't come across that error in another branch/PR..I know there's logic to retry on that error code, i'll double-check if there's something different in this one

**Comment by anGie44** [comment]
Output of updated acceptance tests:
```
--- PASS: TestAccS3BucketServerSideEncryptionConfiguration_disappears (34.58s)
--- PASS: TestAccS3BucketServerSideEncryptionConfiguration_ApplySSEByDefault_KMS (38.80s)
--- PASS: TestAccS3BucketServerSideEncryptionConfiguration_ApplySEEByDefault_AES256 (38.80s)
--- PASS: TestAccS3BucketServerSideEncryptionConfiguration_basic (38.83s)
--- PASS: TestAccS3BucketServerSideEncryptionConfiguration_ApplySSEByDefault_KMSWithMasterKeyArn (39.06s)
--- PASS: TestAccS3BucketServerSideEncryptionConfiguration_ApplySSEByDefault_KMSWithMasterKeyID (39.09s)
--- PASS: TestAccS3BucketServerSideEncryptionConfiguration_BucketKeyEnabled (65.43s)
--- PASS: TestAccS3BucketServerSideEncryptionConfiguration_ApplySSEByDefault_UpdateSSEAlgorithm (65.58s)
--- PASS: TestAccS3BucketServerSideEncryptionConfiguration_ApplySSEByDefault_BucketKeyEnabled (65.71s)
```

**Pull request merged**